includes/Classifai/Providers/Azure/TextToSpeech.php

```diff
@@ -0,0 +1,553 @@
+<?php
+/**
+ * IBM Watson NLU
+ */
+
+namespace Classifai\Providers\Azure;
+
+use Classifai\Providers\Provider;
+use Classifai\Blocks;
+use WP_Http;
+
+use function Classifai\get_post_types_for_language_settings;
+
+class TextToSpeech extends Provider {
+
+	const API_PATH = 'cognitiveservices/v1';
+
+	const SYNTHESIZE_SPEECH_KEY = '_classifai_synthesize_speech';
+
+	const AUDIO_ID_KEY = '_classifai_post_audio_id';
+
+	const AUDIO_TIMESTAMP_KEY = '_classifai_post_audio_timestamp';
+
+	/**
+	 * Watson NLU constructor.
+	 *
+	 * @param string $service The service this class belongs to.
+	 */
+	public function __construct( $service ) {
+		parent::__construct(
+			'Microsoft Azure',
+			'Text to Speech',
+			'azure_text_to_speech',
+			$service
+		);
+	}
+
+	/**
+	 * Enqueue the editor scripts.
+	 */
+	public function enqueue_editor_assets() {
+		global $post;
+		wp_enqueue_script(
+			'classifai-editor', // Handle.
+			CLASSIFAI_PLUGIN_URL . 'dist/editor.js',
+			array( 'wp-blocks', 'wp-i18n', 'wp-element', 'wp-editor', 'wp-edit-post' ),
+			CLASSIFAI_PLUGIN_VERSION,
+			true
+		);
+
+		if ( empty( $post ) ) {
+			return;
+		}
+
+		wp_enqueue_script(
+			'classifai-gutenberg-plugin',
+			CLASSIFAI_PLUGIN_URL . 'dist/gutenberg-plugin.js',
+			array( 'lodash', 'wp-blocks', 'wp-i18n', 'wp-element', 'wp-editor', 'wp-edit-post', 'wp-components', 'wp-data', 'wp-plugins' ),
+			CLASSIFAI_PLUGIN_VERSION,
+			true
+		);
+
+		wp_localize_script(
+			'classifai-gutenberg-plugin',
+			'classifaiTextToSpeechData',
+			[
+				'supportedPostTypes' => \Classifai\get_supported_post_types_for_azure_text_to_speech(),
+				'noPermissions'      => ! is_user_logged_in() || ! current_user_can( 'edit_post', $post->ID ),
+			]
+		);
+	}
+
+	/**
+	 * Can the functionality be initialized?
+	 *
+	 * @return bool
+	 */
+	public function can_register() {
+		$options = get_option( $this->get_option_name() );
```

Comment:
For consistency with other providers, can we change this:
```suggestion
		$settings = $this->get_settings();
```